narwhal/types/src/primary.rs

```diff
@@ -646,6 +646,8 @@ impl Certificate {
         (weight, pks)
     }
 
+    /// Verifies the validaity of the certificate.
+    /// TODO: Output a different type, similar to Sui VerifiedCertificate.
```

Comment:
Very good idea and quite necessary I would say given that now the verification checks are happening outside of core.